src/functionalTest/java/uk/gov/hmcts/reform/sendletter/FunctionalTestSuite.java

```diff
@@ -0,0 +1,197 @@
+package uk.gov.hmcts.reform.sendletter;
+
+import com.google.common.collect.ImmutableMap;
+import com.warrenstrange.googleauth.GoogleAuthenticator;
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.sftp.RemoteFile;
+import net.schmizz.sshj.sftp.SFTPClient;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import java.util.zip.ZipInputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RunWith(SpringRunner.class)
+@SuppressWarnings("PMD.AbstractClassWithoutAbstractMethod")
+@TestPropertySource("classpath:application.properties")
+public abstract class FunctionalTestSuite {
+
+    protected static final String SMOKE_TEST_LETTER_TYPE = "smoke_test";
+
+    @Value("${s2s-url}")
+    protected String s2sUrl;
+
+    @Value("${s2s-name}")
+    protected String s2sName;
+
+    @Value("${s2s-secret}")
+    protected String s2sSecret;
+
+    @Value("${producer-url}")
+    protected String producerUrl;
+
+    @Value("${ftp-hostname}")
+    protected String ftpHostname;
+
+    @Value("${ftp-port}")
+    protected Integer ftpPort;
+
+    @Value("${ftp-fingerprint}")
+    protected String ftpFingerprint;
+
+    @Value("${ftp-target-folder}")
+    protected String ftpTargetFolder;
+
+    @Value("${ftp-user}")
+    protected String ftpUser;
+
+    @Value("${ftp-private-key}")
+    protected String ftpPrivateKey;
+
+    @Value("${ftp-public-key}")
+    protected String ftpPublicKey;
```

Comment:
all this can be replaced with `FtpConfigProperties`

Reply:
This configuration is not exactly the same - there are differences, e.g. reports folder (doesn't exist for tests), target folder vs smoke test target folder (both exist in app's config, tests only care about their folder). And then there are some other non-ftp config properties that have to be defined. This configuration is specific for tests and even if there is a subset of properties common for app and tests, I'd rather keep them independent.

What I could do is extract those properties into a separate (still test-specific) config class. What do you think?

Reply:
i thought test specific properties are loaded in the current scope so you can simply override e.g. `ftp.public-key` and then just include the ftp config in the suite. would make less of a scroll in the file. any other specific properties are ok. just ftp is quite huge hence the separate properties file.

if it's a pain and/or doesn't actually fit the needs of a functional test set up - i'm happy to ignore

Reply:
I'd prefer to avoid reusing FtpConfigProperties, so that I wouldn't have to define some properties just so that they're not missing (otherwise the framework would fail) and allow the actual FtpConfigProperties (which are not strictly ftp-specific) to be maintained/extended independently.

I can still make a separate config class, so that the suite class is a bit shorter, if you prefer.